### ISSUE TITLE
How to get notified only after app has been launched? Bug?

### ISSUE THREAD

**Issue by CyberMew** [react-native-share/react-native-share#833]
## Ask your Question

I want to delete a file that's temporary stored in the cache folder after calling `Share.open` on Android to launch Gmail (no issues on iOS).

However once I do that, all the `.then.then.finally` gets called even _before_ the file is attached to Gmail. There is no chance for me to do a cleanup.

The only time I can track is when the user regrets calling `Share.open` and dismisses the popup sheet (i.e. before selecting Gmail), resulting in a `.catch.finally` situation and I can cleanup in `.catch`.

Is there no way to do that?

Edit:

Just to be clear, on iOS, when I have shared the files successfully and the popup sheet has fully disappear, then would the `.then.then.finally` be called. This behaviour _does not_ match the Android behaviour. Is this a bug?

**Comment by CyberMew** [comment]
tl;dr:

iOS sharing completion called `finally` 👍. 

Android sharing halfway called finally - resulting me in unable to delete file because behaviour is uncertain 👎

**Comment by MateusAndrade** [comment]
Hi @CyberMew, sorry for the long time to answer. You are saying that you are able to detect whether the user shared or not on iOS. But on Android, the finally is thrown whenever the user completes or not the action?

**Comment by CyberMew** [comment]
Hi @MateusAndrade, yes. Once gmail is selected, the `try catch finally` completes, which is inconsistent with iOS. I am depending on `finally` only be called when the file has been successfully shared as I am deleting the file over there. For iOS this is working great, but for Android it calls `finally` way too early and hence my file is deleted before it attaches to gmail. There is no way at all to track if the sharing action has indeed been fully completed successfully or not, for android.

**Comment by CyberMew** [comment]
Yes it’s affecting me but I have no idea how to fix it and this ticket is apparently not considered at all so I’m not sure what’s happening.

There is no consistency at all between the 2 platforms in terms of the processing flow and I’m surprised no one is chipping in (or maybe they did in other tickets).

**Comment by MateusAndrade** [comment]
Hi @CyberMew, sorry for the Spam caused by the stale-bot. I'm re-opening this and going to have a proper look at Android using our example.
